scrapy/utils/deprecate.py

```diff
@@ -121,3 +122,13 @@ def _clspath(cls, forced=None):
     if forced is not None:
         return forced
     return '{}.{}'.format(cls.__module__, cls.__name__)
+
+
+def update_classpath(path):
+    """Update a deprecated path from an object with its new location.
+
+    This function tries to load the object based on the given path, hence it'll
+    only work if the old path is still supported.
+    """
+    obj = load_object(path)
```

Comment:
calling `load_object()` is a bad idea not only because it triggers an import but because it brings all the side cases of skipping the error trapping logic of https://github.com/scrapy/scrapy/blob/master/scrapy/middleware.py#L31-L45